ddp-workspace/projects/ddp-dsm-ui/src/app/FON/components/paginator/paginator.component.ts

```diff
@@ -0,0 +1,143 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  Input,
+  OnChanges,
+  Output,
+  SimpleChanges
+} from '@angular/core';
+
+
+interface pageProps {
+  from: number;
+  to: number;
+}
+
+enum SHIFT_PAGE {
+  START = 'START',
+  END = 'END',
+  ONE_MORE = 'ONE_MORE',
+  ONE_LESS = 'ONE_LESS'
+}
+
+/* Defaults */
+const PAGE_SIZE_OPTIONS = [10, 25, 50];
+const ROWS_PER_PAGE = 10;
+const CURRENT_PAGE_INDEX = 1;
+const VISIBLE_PAGES = 3;
+
+
+@Component({
+  selector: 'app-paginator',
+  templateUrl: './paginator.component.html',
+  styleUrls: ['./paginator.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class PaginatorComponent implements OnChanges {
+  @Input() totalCount: number;
+  @Input() rowsPerPage = 10;
+  @Input() currentPageIndex: number;
+  @Input() pageSizeOptions: number[];
+  @Input() visiblePages: number;
+
+  @Output() pageChanged: EventEmitter<pageProps> = new EventEmitter();
+
+  constructor() {}
+
+  ngOnChanges(changes: SimpleChanges): void {
+    this.setDefaultParams();
+    this.generatePagesArray();
```

Comment:
What is the purpose of that method ? 
Seems like something is forgotten here.

Reply:
When you refresh the page, or just navigate through pages, each time request is made to the backend, therefore we have to set the current state of the paginator when data comes as we are not initializing the paginator component every time (especially when we just move from one page to the other), otherwise, OnInit hook would be fine.

Reply:
Unused property is deleted

Reply:
Sorry, I did not get it.
`this.generatePagesArray();` here returns an array of numbers (page numbers actually).
But is is never/nowhere used.  
No value/property is changed by that returned pageArray.
Is not it?

Reply:
I just had a look at this again and realized that using generatePagesArray() function call in that hook is redundant. I think, I had left it there because of previous implementation that I modified later and I forgot to delete this.

Reply:
But we can leave the setDefaultParams() function at this moment. Later, after switching to setters, I will remove that with other changes. Thanks for noticing that.

Reply:
Removed!